spec/datadog/appsec/contrib/rails/integration_test_spec.rb

```diff
@@ -0,0 +1,284 @@
+# typed: ignore
+
+require 'datadog/tracing/contrib/rails/rails_helper'
+
+require 'datadog/appsec'
+
+RSpec.describe 'Rails integration tests' do
+  include Rack::Test::Methods
+
+  let(:sorted_spans) do
+    chain = lambda do |start|
+      loop.with_object([start]) do |_, o|
+        # root reached (default)
+        break o if o.last.parent_id == 0
+
+        parent = spans.find { |span| span.span_id == o.last.parent_id }
+
+        # root reached (distributed tracing)
+        break o if parent.nil?
+
+        o << parent
+      end
+    end
+    sort = ->(list) { list.sort_by { |e| chain.call(e).count } }
+    sort.call(spans)
+  end
+
+  let(:rack_span) { sorted_spans.reverse.find { |x| x.name == Datadog::Tracing::Contrib::Rack::Ext::SPAN_REQUEST } }
+
+  let(:appsec_enabled) { true }
+  let(:tracing_enabled) { true }
+
+  before do
+    Datadog.configure do |c|
+      c.tracing.enabled = tracing_enabled
+      c.tracing.instrument :rails
+
+      c.appsec.enabled = appsec_enabled
+      c.appsec.instrument :rails
+
+      # TODO: test with c.appsec.instrument :rack
+    end
+  end
+
+  context 'for an application' do
+    include_context 'Rails test application'
+
+    let(:controllers) { [controller] }
+
+    let(:controller) do
+      stub_const(
+        'TestController',
+        Class.new(ActionController::Base) do
+          # skip CSRF token check for non-GET requests
+          begin
+            if respond_to?(:skip_before_action)
+              skip_before_action :verify_authenticity_token
+            else
+              skip_before_filter :verify_authenticity_token
+            end
+          rescue ArgumentError # :verify_authenticity_token might not be defined
+            nil
+          end
+
+          def success
+            render plain: 'ok'
+          end
+        end
+      )
+    end
+
+    let(:triggers) do
+      json = trace.send(:meta)['_dd.appsec.json']
+
+      JSON.parse(json).fetch('triggers', []) if json
+    end
+
+    let(:span) { rack_span }
+
+    shared_examples 'a GET 200 span' do
+      it { expect(span.get_tag('http.method')).to eq('GET') }
+      it { expect(span.get_tag('http.status_code')).to eq('200') }
+      it { expect(span.status).to eq(0) }
+
+      context 'with appsec disabled' do
+        let(:appsec_enabled) { false }
+
+        it { expect(span.get_tag('http.method')).to eq('GET') }
+        it { expect(span.get_tag('http.status_code')).to eq('200') }
+        it { expect(span.status).to eq(0) }
+      end
+    end
```

Comment:
I noticed that the rack integration tests don't have test for appsec being disabled as we do for rails here. Would it make sense to expand those tests? (Or perhaps leave a `pending`/`skip` for now?)

Reply:
Good catch, it should be tested as well.

Reply:
Actually they had it on the appsec event+tag shared examples, they were just absent of the regular sanity-check tracing ones. Anyway, added for consistency.